ci_framework/roles/cifmw_cephadm/tasks/dashboard/dashboard.yml

```diff
@@ -0,0 +1,117 @@
+      changed_when: false
+  when: cifmw_cephadm_dashboard_crt | length > 0 and
+        cifmw_cephadm_dashboard_key | length > 0
+
+- include_tasks: configure_dashboard_backends.yml
+  with_items: "{{ groups['edpm'] }}"
+  vars:
+    dashboard_backend: '{{ item }}'
+
+- name: Restart the Ceph dashboard
+  become: true
+  block:
+    - name: disable mgr dashboard module (restart)
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} mgr module disable dashboard"
+    - name: enable mgr dashboard module (restart)
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} mgr module enable dashboard"
+
+- name: Configure Monitoring Stack
+  become: true
+  vars:
+    grafana_host:  "{{ groups['edpm'][0] }}"
+  block:
+    - name: get Grafana instance address
+      ansible.builtin.set_fact:
+        grafana_server_addr: "{{ hostvars[grafana_host][all_addresses] | ansible.utils.ipaddr(cifmw_cephadm_monitoring_network) | first }}"
+    - name: set grafana api user
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} dashboard set-grafana-api-username {{ cifmw_cephadm_grafana_admin_user }}"
+    - name: set grafana api password
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} dashboard set-grafana-api-password -i -"
+      args:
+        stdin: "{{ cifmw_cephadm_grafana_admin_password }}"
+        stdin_add_newline: no
+    - name: disable ssl verification for grafana
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} dashboard set-grafana-api-ssl-verify False"
+      changed_when: false
+      when:
+        - cifmw_cephadm_dashboard_crt | length > 0 and
+          cifmw_cephadm_dashboard_key | length > 0
+
+    - name: set alertmanager host
+      ansible.builtin.command: |
+        {{ cifmw_cephadm_ceph_cli }} dashboard set-alertmanager-api-host http://{{ grafana_server_addr }}:9093
+    - name: set prometheus host
+      ansible.builtin.command: |
+        {{ cifmw_cephadm_ceph_cli }} dashboard set-prometheus-api-host \
+        http://{{ grafana_server_addr }}:{{ cifmw_cephadm_prometheus_port }}
+    - name: config grafana api url
+      ansible.builtin.command: |
+        {{ cifmw_cephadm_ceph_cli }} dashboard set-grafana-api-url \
+        http://{{ grafana_server_addr }}:{{ cifmw_cephadm_grafana_port }}
+
+- name: Restart the Ceph dashboard
+  become: true
+  block:
+    - name: disable mgr dashboard module (restart)
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} mgr module disable dashboard"
+    - name: enable mgr dashboard module (restart)
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} mgr module enable dashboard"
+
+- name: 20 second pause for dashboard services to be up and running.
+  ansible.builtin.pause:
+    seconds: 20
+
+- name: Try connecting to dashboard service
+  ansible.builtin.command: "curl -k http://{{ grafana_server_addr }}:{{ cifmw_cephadm_dashboard_port }}"
```

Comment:
this part can be extended to perform a more complex query against the dashboard. I understand the current `curl` is used to make sure the dashboard is working on the specified endpoint, but we might want to extend this to perform  more complex queries against the endpoint